QLearningMaze/Maze.Core/Mazes/MazeBase.cs

```diff
@@ -104,38 +113,78 @@ protected virtual void CreateMazeStates()
 
             Console.WriteLine("Creating Maze States (Observation Space)");
 
-            int[][] mazeNextStates = new int[NumberOfStates][];
+            int[][] mazeStatesActions = new int[NumberOfStates][];
 
             for (int i = 0; i < NumberOfStates; ++i)
             {
-                mazeNextStates[i] = new int[NumberOfStates];
+                mazeStatesActions[i] = new int[5];
```

Comment:
In case this can expand to include diagonals later, use the number of values in the Actions enum instead of a literal